koji_containerbuild/plugins/builder_containerbuild.py

```diff
@@ -506,6 +510,9 @@ def createContainer(self, src=None, target_info=None, arches=None,
         if not arches:
             raise ContainerError("arches aren't specified")
 
+        if signing_intent and compose_ids:
+            raise ContainerError("signing_intent used with compose_ids")
```

Comment:
This doesn't seem like the right exception to raise. Probably `koji.BuildError` is better for this. I see `ContainerError` is also raised if arches is not specified -- that also looks wrong to me.

Reply:
Good catch! It looks like this is also how it's done in RPM land:

```python
def getArchList(self, build_tag, h, extra=None):                                  
    # get list of arches to build for                                             
    buildconfig = self.session.getBuildConfig(build_tag, event=self.event_id)     
    arches = buildconfig['arches']                                                
    if not arches:                                                                
        #XXX - need to handle this better                                         
        raise koji.BuildError("No arches for tag %(name)s [%(id)s]" % buildconfig)
    tag_archlist = [koji.canonArch(a) for a in arches.split()]                    
    self.logger.debug('arches: %s' % arches)                                      
    if extra:                                                                     
        self.logger.debug('Got extra arches: %s' % extra)                         
        arches = "%s %s" % (arches,extra)                                         
    archlist = arches.split()                                                     
```
https://pagure.io/koji/blob/master/f/builder/kojid#_958